src/main/kotlin/com/zepben/evolve/database/sqlite/metrics/MetricsDataSourceWriter.kt

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2024 Zeppelin Bend Pty Ltd
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+package com.zepben.evolve.database.sqlite.metrics
+
+import com.zepben.evolve.database.sqlite.common.SchemaUtils
+import com.zepben.evolve.database.sqlite.common.TableVersion
+import com.zepben.evolve.database.sqlite.extensions.configureBatch
+import com.zepben.evolve.metrics.IngestionJob
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import java.sql.Connection
+import javax.sql.DataSource
+
+/**
+ * Class for writing metrics to an arbitrary datasource.
+ */
+class MetricsDataSourceWriter @JvmOverloads constructor(
+    private val dataSource: DataSource,
```

Comment:
todo: This is public API and parameter `dataSource` isn't documented. I have excluded `databaseTables` from this as it is for test injection as per the other comment.